app/views/posts/index.html.erb

```diff
@@ -1 +1,36 @@
-<h1>this is the main page of post</h1>
\ No newline at end of file
+<h2><%= @user.name %>'s Posts page</h2>
+<div class="user-card">
+  <div class="user-info">
+    <h3><%= @user.name %></h3>
+    <p>Number of posts: <%= @user.posts_counter || 0 %></p>
+  </div>
+  <img class="user-photo" src="<%= @user.photo %>" alt="individual user photo">
+</div>
+
+<div>
+  <% @posts.each do |post| %>
+    <%= link_to user_post_path(user_id: @user.id, id: post.id), class: "user-card-link" do %>
+    <div class="card">
+     <h2><%= post.title %></h2>
+     <p><%= post.text %></p>
+     <div class ="like-com">
+      <p>Likes: <%= post.likes.count %></p>
+      <p>Comments: <%= post.comments.count %></p>
+    </div>
+  <% end %>
+
+   </div>
+   <% if post.comments.any? %>
+      <div class="card">
+        <ul class="ul">
+          <% post.comments.each do |comment| %>
+            <li><%= comment.text %></li>
+          <% end %>
+        </ul>
```

Comment:
- [ ] As instructed in the project requirement, kindly ensure that only the five most recent comments for a given post are displayed in the  UI. 👍